dd-sdk-android-androidx/src/main/kotlin/com/datadog/android/androidx/tracking/JetpackExtraAttributesProvider.kt

```diff
@@ -0,0 +1,42 @@
+package com.datadog.android.androidx.tracking
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import com.datadog.android.rum.tracking.ViewAttributesProvider
+
+/**
+ * Provides extra attributes for the touch target View.
+ * Those special attributes are specifically related with Jetpack components.
+ * Examples:
+ * <ul>
+ *     <li> If the parent of the target view is a RecyclerView it will add the position
+ *     of the target inside the adapter as a special attribute </li>
+ * </ul>
+ * @see [TAG_TARGET_ADAPTER_POSITION]
+ */
+object JetpackExtraAttributesProvider : ViewAttributesProvider {
+
+    internal const val TAG_TARGET_ADAPTER_POSITION = "target.adapterPosition"
+
+    override fun addExtraAttributes(
+        view: View,
+        attributes: MutableMap<String, Any?>
+    ) {
+        // traverse the target parents
+        var parent = view.parent
+        var child: View? = view
+        while (parent != null) {
+            if (parent is RecyclerView && child != null && isDirectChildOfRecyclerView(child)) {
+                val positionInAdapter = parent.getChildAdapterPosition(child)
+                attributes[TAG_TARGET_ADAPTER_POSITION] = positionInAdapter
+                break
+            }
+            child = parent as? View
+            parent = parent.parent
+        }
+    }
+
+    private fun isDirectChildOfRecyclerView(child: View): Boolean {
```

Comment:
Is this really necessary ? Like in what case are there views as children of a recyclerview but without a RecyclerView.LayoutParams ?

Reply:
I really wanted to be safe here. If by any chance that direct descendant doesn't have the RecyclerView.LayoutParams it will crash when trying to case those. Not sure here but I wanted to be better safe than sorry later.